### PR TITLE
owls-77750 dedicated ns

### PR DESCRIPTION
**Pull request by doxiao**
This PR contains the project for owls-77750.
The goal is to support a new mode where the operator runs in a dedicated namespace where all the domains that it manages are also running. The feature is turned on by a new helm chart setting "dedicated", which is false by default. 
In the initial solution, the customer has to manually create the domain CRD object prior to install an operator.  We have generated a domain CRD yaml file in the directory kubernetes/charts/weblogic-domain directory for customer to use.

### CODE REVIEW

**Comment by doxiao** [comment]
> There are a fair number of code changes, but no corresponding tests for the new functionality. The only test edit appears to be have been required to pass, as a default changed.
> 
> Please write unit tests for new and changed functionality. I am more than happy to work with you to create them.

**Comment by doxiao** [comment]
Sorry, accidentally closed the RP.

**Comment by doxiao** [comment]
Doc changes are in. Russ and I worked together to add more unit test cases.
The PR is ready to be reviewed.

File: docs-source/content/security/rbac.md [hunk 1]
```diff
@@ -132,17 +132,18 @@ the following `ClusterRoleBinding` entries are mapped to a `ClusterRole` grantin
...
+| Operator `discovery` | Kubernetes `system:discovery` | **See**: [Kubernetes Discovery Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#discovery-roles) | [^3] |
+| Operator `auth-delegator` | Kubernetes `system:auth-delegator` | **See**: [Kubernetes Component Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#other-component-roles) | [^3] |
 
 
 [^1]: The binding is assigned to the operator `ServiceAccount`.
 [^2]: The binding is assigned to the operator `ServiceAccount`
       in each namespace listed with the `domainNamespaces` setting.
       The `domainNamespaces` setting contains the list of namespaces
       that the operator is configured to manage.
+[^3]: The binding is assigned to the operator `ServiceAccount`. In additional, by default (when the `dedicated` value is set to `false`), those are ClusterRole and ClusterRoleBindings. However, if the `dedicated` value is set to `true`,  those will become Roles and RoleBindings in the namespace of the operator.
```
Comment (markxnelson) [inline docs-source/content/security/rbac.md]: this sentence needs to be reworded to make it less confusing.  ClusterRole, etc., should be backticked

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md [hunk 2]
```diff
@@ -94,6 +94,23 @@ Example:
 serviceAccount: "weblogic-operator"
 ```
 
+##### `dedicated`
+
+Specifies if this operator will only manage WebLogic Domains in the same namespace in which the operator itself is deployed.  If set as `true`, then the `domainNamespaces` value below is ignored.
```
Comment (markxnelson) [inline docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: as -> to

Comment (markxnelson) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: remove "below"

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md [hunk 3]
```diff
@@ -94,6 +94,23 @@ Example:
...
+Specifies if this operator will only manage WebLogic Domains in the same namespace in which the operator itself is deployed.  If set as `true`, then the `domainNamespaces` value below is ignored. 
+
+Defaults to `false`.
+
+Example:
+```
+dedicated: false
+```
+
+In the `dedicated` mode, the operator will most likely not have permissions to access the cluster-scoped Kubernetes resources such as CustomResourceDefinitions, PersistentVolumes, and Namespaces. In those situations, the operator may skip some of its operations, such as verifying the WebLogic Domain CustomResoruceDefinition domains.weblogic.oracle (and creating it when it is absent), watching namespace events, and cleaning up PersistentVolumes as part of deleting a domain.
```
Comment (markxnelson) [inline docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: don't say "will most likely not have"  
say "does not require"

Comment (markxnelson) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: backtick things like CustomResourceDefinitions

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md [hunk 4]
```diff
@@ -94,6 +94,23 @@ Example:
...
+
+Example:
+```
+dedicated: false
+```
+
+In the `dedicated` mode, the operator will most likely not have permissions to access the cluster-scoped Kubernetes resources such as CustomResourceDefinitions, PersistentVolumes, and Namespaces. In those situations, the operator may skip some of its operations, such as verifying the WebLogic Domain CustomResoruceDefinition domains.weblogic.oracle (and creating it when it is absent), watching namespace events, and cleaning up PersistentVolumes as part of deleting a domain. 
+
+{{% notice note %}}
+It is the responsibility of the administrators to make sure that the required CustomResourceDefinition domains.weblogic.oracle is deployed in the Kubernetes cluster before the operator is installed. A YAML file for the CRD (`domain-crd.yaml`) can be found in `kubernetes/crd` directory of the operator project.
```
Comment (markxnelson) [inline docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: administrator singular

Comment (markxnelson) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: backtick names of objects

Comment (markxnelson) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: make a hyperlink to the actual file

File: docs/charts/index.yaml [hunk 5]
```diff
@@ -1,53 +1,53 @@
...
+  - created: "2020-01-15T20:11:19.500388682Z"
     description: Helm chart for configuring the WebLogic operator.
-    digest: b36bd32083f67453a62d089a2c09ce38e6655d88ac8a7b38691230c55c40e672
+    digest: d358882813647586f6137061de441c4b89c775dcda6653e8cc6e6cf2222cb48b
     name: weblogic-operator
     urls:
     - https://oracle.github.io/weblogic-kubernetes-operator/charts/weblogic-operator-2.4.0.tgz
     version: 2.4.0
-  - created: "2019-11-15T20:34:35.37493-05:00"
+  - created: "2020-01-15T20:11:19.499534741Z"
```
Comment (markxnelson) [inline docs/charts/index.yaml]: why are you changing the old ones?

File: kubernetes/charts/weblogic-operator/templates/_domain-namespaces.tpl [hunk 6]
```diff
@@ -1,10 +1,17 @@
...
 {{- define "operator.domainNamespaces" }}
+{{- if not .dedicated }}
 {{- $args := include "utils.cloneDictionary" . | fromYaml -}}
 {{- range $key := $args.domainNamespaces -}}
 {{-   $ignore := set $args "domainNamespace" $key -}}
 {{-   include "operator.operatorRoleBindingNamespace" $args -}}
 {{- end }}
+{{- else }}
+{{- $args := include "utils.cloneDictionary" . | fromYaml -}}
+{{- $key := .Release.Namespace -}}
```
Comment (markxnelson) [inline kubernetes/charts/weblogic-operator/templates/_domain-namespaces.tpl]: is this right? looks wrong

File: kubernetes/charts/weblogic-operator/values.yaml [hunk 7]
```diff
@@ -7,8 +7,13 @@
 # If not specified, the the operator will use the operator's namespace's 'default' service account.
 serviceAccount: "default"
 
-# domainNamespaces specifies list of WebLogic Domain namespaces which the operator manages.
-# The customer is responsible for creating these namespaces.
+# dedicated specifies if this operator will manage WebLogic Domains only in the same namespace in
+# which the operator itself is deployed.  If set as 'true', then the 'domainNamespaces' value below
```
Comment (markxnelson) [inline kubernetes/charts/weblogic-operator/values.yaml]: as -> to

File: kubernetes/charts/weblogic-operator/values.yaml [hunk 8]
```diff
@@ -7,8 +7,13 @@
...
+# is ignored.
+dedicated: false
+
+# domainNamespaces specifies list of WebLogic Domain namespaces which the operator manages. This value
+# is ignored if 'dedicated' is set as 'true'. The customer is responsible for creating these namespaces.
```
Comment (markxnelson) [inline kubernetes/charts/weblogic-operator/values.yaml]: as -> to

Comment (markxnelson) [reply kubernetes/charts/weblogic-operator/values.yaml]: customer -> administrator

File: kubernetes/src/test/java/oracle/kubernetes/operator/helm/HelmOperatorValuesTest.java [hunk 9]
```diff
@@ -487,6 +487,50 @@ public void whenCreatedFromMapWithExternalDebugHttpPort_hasSpecifiedValue() {
     assertThat(values.getExternalDebugHttpPort(), equalTo(stringValue));
   }
 
+  // --------------- dedicated
+
+  @Test
+  public void whenDedicatedTrue_createdMapContainsValue() {
+    operatorValues.dedicated("true");
+
```
Comment (markxnelson) [inline kubernetes/src/test/java/oracle/kubernetes/operator/helm/HelmOperatorValuesTest.java]: why all the empty lines?

File: operator/src/main/java/oracle/kubernetes/operator/Main.java [hunk 10]
```diff
@@ -578,6 +595,17 @@ public NextAction apply(Packet packet) {
...
 
+  /**
+   * Adds the specified namespace to the list of started namespaces, returning true if it was not
+   * already present.
+   * @param ns the namespace name
+   * @return false if the namespace had already been started
+   */
+  static boolean addStartedNamespace(String ns) {
+    AtomicBoolean a = isNamespaceStarted.computeIfAbsent(ns, (key) -> new AtomicBoolean(false));
+    return !a.getAndSet(true);
```
Comment (markxnelson) [inline operator/src/main/java/oracle/kubernetes/operator/Main.java]: this line seems overly complicated and prone to future update errors

File: operator/src/main/java/oracle/kubernetes/operator/StartupControl.java [hunk 11]
```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (markxnelson) [inline operator/src/main/java/oracle/kubernetes/operator/StartupControl.java]: Do not include "All rights reserved" in any new files -- new rules....

File: operator/src/main/java/oracle/kubernetes/operator/StartupControl.java [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+import oracle.kubernetes.operator.helpers.AuthorizationProxy.Resource;
+import oracle.kubernetes.operator.helpers.CrdHelper;
+import oracle.kubernetes.operator.helpers.HealthCheckHelper;
+import oracle.kubernetes.operator.helpers.KubernetesVersion;
+import oracle.kubernetes.operator.logging.LoggingFacade;
+import oracle.kubernetes.operator.logging.LoggingFactory;
+import oracle.kubernetes.operator.logging.MessageKeys;
+import oracle.kubernetes.operator.work.Step;
+
+public class StartupControl {
```
Comment (markxnelson) [inline operator/src/main/java/oracle/kubernetes/operator/StartupControl.java]: javadoc???

File: operator/src/main/java/oracle/kubernetes/operator/helpers/CrdHelper.java [hunk 13]
```diff
@@ -43,6 +52,31 @@
...
+    }
+
+    String outputFileName = args[0];
+
+    Path outputFilePath = Paths.get(outputFileName);
+    CrdContext context = new CrdContext(null, null);
+
+    try (Writer writer = Files.newBufferedWriter(outputFilePath)) {
+      writer.write(
+          "# Copyright (c) 2019, Oracle Corporation and/or its affiliates.  All rights reserved.\n"
```
Comment (markxnelson) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/CrdHelper.java]: 2020

File: operator/src/main/java/oracle/kubernetes/operator/helpers/HealthCheckHelper.java [hunk 14]
```diff
@@ -21,151 +26,190 @@
...
   private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Operator", "Operator");
 
-  private static final Map<AuthorizationProxy.Resource, AuthorizationProxy.Operation[]>
+  private static final Map<Resource, Operation[]>
       namespaceAccessChecks = new HashMap<>();
-  private static final Map<AuthorizationProxy.Resource, AuthorizationProxy.Operation[]>
+  private static final Map<Resource, Operation[]>
       clusterAccessChecks = new HashMap<>();
 
   // Note: this list should match the RBAC or ABAC policies contained in the YAML script
```
Comment (markxnelson) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/HealthCheckHelper.java]: we don't support ABAC...

File: operator/src/main/resources/Operator.properties [hunk 15]
```diff
@@ -163,6 +163,15 @@ WLSKO-0154=Job {0} failed due to reason: DeadlineExceeded. \
 WLSKO-0155=Unexpected exception, {0}, while parsing introspect job log [{1}].
 WLSKO-0156=Access denied for operator service account for operation {0} on resource {1} in namespace {2}.
 WLSKO-0157=Domain {0} is not valid: {1}
+WLSKO-0158=Expected CustomResourceDefinition domains.weblogic.oracle not found. But the operator's service account \
+  does not have the permissions to {0}. One solution is to create or update the CRD and  \
```
Comment (markxnelson) [inline operator/src/main/resources/Operator.properties]: remove "one solution is to"

File: operator/src/main/resources/Operator.properties [hunk 16]
```diff
@@ -163,6 +163,15 @@ WLSKO-0154=Job {0} failed due to reason: DeadlineExceeded. \
...
+  re-install the operator. 
+WLSKO-0159=Skipped the verification of the CustomResourceDefinition domains.weblogic.oracle because \
+  the operator's service account does not have the permissions to access the existing CustomResourceDefinition.
```
Comment (markxnelson) [inline operator/src/main/resources/Operator.properties]: the permissions -> permission

File: operator/src/main/resources/Operator.properties [hunk 17]
```diff
@@ -163,6 +163,15 @@ WLSKO-0154=Job {0} failed due to reason: DeadlineExceeded. \
...
+WLSKO-0160=Verifying that operator service account can perform operation {1} on required resource {0}.
+WLSKO-0161=Skipped the creation of the namespace watcher because the operator's service account \
+  does not have the permissions to access the namespaces in the Kubernetes cluster.\
```
Comment (markxnelson) [inline operator/src/main/resources/Operator.properties]: the permissions -> permission

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java [hunk 18]
```diff
@@ -0,0 +1,256 @@
+// Copyright (c) 2020, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (markxnelson) [inline operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: remove "All rights reserved"

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+import static oracle.kubernetes.utils.LogMatcher.containsWarning;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
+
+public class StartupControlTest {
+  private static final KubernetesVersion VERSION = new KubernetesVersion(1, 8);
```
Comment (markxnelson) [inline operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: 1,8? seems old, should we update?

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java [hunk 20]
```diff
@@ -0,0 +1,256 @@
...
+  public void whenOperatorDedicatedEmptyString_isDedicatedReturnsFalse() {
+    environmentVariables.put(StartupControl.OPERATOR_DEDICATED_ENV, "");
+    
+    assertThat(startupControl.isDedicated(), is(false));
+  }
+
+  @Test
+  public void whenOperatorDedicatedNotTrue_isDedicatedReturnsFalse() {
+    environmentVariables.put(StartupControl.OPERATOR_DEDICATED_ENV, "zork");
+
```
Comment (markxnelson) [inline operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: why all the blank lines?

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java (hunk 18 above)
Comment (russgold) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: They all say that. Is this a change?

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java (hunk 19 above)
Comment (russgold) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: We absolutely should. There is still logic that determines whether Rules Review is supported, even though we no longer support versions before 1.8. I don't think it should be done as part of this PR, though.

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java (hunk 20 above)
Comment (russgold) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: Do you mean inside the test methods? That's a common convention to separate individual test setup, the invocation being tested, and the assertion of the expected result.

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java (hunk 19 above)
Comment (rjeberhard) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: I have a PR coming later in the week that addresses this.  There is another set of JIRA's about removing unnecessary privileges and also working in a "low privilege" environment and I used those as an opportunity to clean-up this code.

Comment (russgold) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: You'll need to adjust the code in this PR if you do.

File: docs/charts/index.yaml (hunk 5 above)
Comment (russgold) [reply docs/charts/index.yaml]: I believe this is auto-generated.

Comment (doxiao) [reply docs/charts/index.yaml]: yes, this file gets updated automatically.

File: kubernetes/charts/weblogic-operator/templates/_domain-namespaces.tpl (hunk 6 above)
Comment (doxiao) [reply kubernetes/charts/weblogic-operator/templates/_domain-namespaces.tpl]: If the dedicated is set to true, we ignore the domainNamespaces setting, and create the namespaced roles and rolebindings for the operator's namespace only.

File: operator/src/main/java/oracle/kubernetes/operator/Main.java (hunk 10 above)
Comment (doxiao) [reply operator/src/main/java/oracle/kubernetes/operator/Main.java]: This is just a refactoring of the old code in line #567.  The pattern is used in other places.

File: operator/src/main/java/oracle/kubernetes/operator/helpers/CrdHelper.java (hunk 13 above)
Comment (doxiao) [reply operator/src/main/java/oracle/kubernetes/operator/helpers/CrdHelper.java]: good catch

File: kubernetes/charts/weblogic-operator/templates/_domain-namespaces.tpl (hunk 6 above)
Comment (doxiao) [reply kubernetes/charts/weblogic-operator/templates/_domain-namespaces.tpl]: this works as expected.

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java (hunk 18 above)
Comment (doxiao) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: @markxnelson The operator runtime build failed if I remove "All rights reserved" part with the following warning.
[WARNING] src/main/java/oracle/kubernetes/operator/StartupControl.java:[1] (header) RegexpHeader: Line does not match expected header line of '^// Copyright \(c\) (\d\d\d\d, )+Oracle Corporation and\/or its affiliates\.  All rights reserved\.$'.

File: docs-source/content/security/rbac.md (hunk 1 above)
Comment (doxiao) [reply docs-source/content/security/rbac.md]: I reworded this sentence and please review.

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md (hunk 2 above)
Comment (doxiao) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: both done

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md (hunk 3 above)
Comment (doxiao) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: done

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md (hunk 4 above)
Comment (doxiao) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: done.

File: operator/src/main/resources/Operator.properties (hunk 16 above)
Comment (doxiao) [reply operator/src/main/resources/Operator.properties]: done

File: operator/src/main/resources/Operator.properties (hunk 17 above)
Comment (doxiao) [reply operator/src/main/resources/Operator.properties]: done.

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java (hunk 18 above)
Comment (doxiao) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: left unchanged.

File: kubernetes/charts/weblogic-operator/values.yaml (hunk 7 above)
Comment (doxiao) [reply kubernetes/charts/weblogic-operator/values.yaml]: done

File: kubernetes/charts/weblogic-operator/values.yaml (hunk 8 above)
Comment (doxiao) [reply kubernetes/charts/weblogic-operator/values.yaml]: both done.

File: operator/src/main/java/oracle/kubernetes/operator/StartupControl.java (hunk 12 above)
Comment (doxiao) [reply operator/src/main/java/oracle/kubernetes/operator/StartupControl.java]: added

**Comment by doxiao** [comment]
> i expected to see some more documentation updates too - especially the security section, the install section, and the openshift \support statement

I added more doc content in the openshift part. I checked other areas and did not see any places where more doc is needed for this.

File: docs-source/content/security/openshift.md [hunk 21]
```diff
@@ -44,3 +44,8 @@ spec:
 For additional information about OpenShift requirements and the operator,
 see the [OpenShift]({{<relref  "/userguide/introduction/introduction.md#openshift">}}) section in the User Guide.
 {{% /notice %}}
+
+#### Using a dedicated namespace
+
+When the user that installs an individual instance of the operator does not have the required privileges to create resources at the Kubernetes cluster level, a dedicated namespace can be used for the operator instance and all WebLogic domains that it manages. For more details of the `dedicated` setting, please refer to [Operator Helm configuration values]({{< relref "/userguide/managing-operators/using-the-operator/using-helm.md#operator-helm-configuration-values" >}}).
```
Comment (rosemarymarano) [inline docs-source/content/security/openshift.md]: and all WebLogic -> and all the WebLogic
For more details of the -> For more details about the

File: docs-source/content/security/rbac.md [hunk 22]
```diff
@@ -132,17 +132,18 @@ the following `ClusterRoleBinding` entries are mapped to a `ClusterRole` grantin
...
+| Operator `discovery` | Kubernetes `system:discovery` | **See**: [Kubernetes Discovery Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#discovery-roles) | [^3] |
+| Operator `auth-delegator` | Kubernetes `system:auth-delegator` | **See**: [Kubernetes Component Roles](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#other-component-roles) | [^3] |
 
 
 [^1]: The binding is assigned to the operator `ServiceAccount`.
 [^2]: The binding is assigned to the operator `ServiceAccount`
       in each namespace listed with the `domainNamespaces` setting.
       The `domainNamespaces` setting contains the list of namespaces
       that the operator is configured to manage.
+[^3]: The binding is assigned to the operator `ServiceAccount`. In addition, the Kubernetes RBAC resources that the operator installation actually set up will be adjusted based on the value of the `dedicated` setting. By default,  the `dedicated` value is set to `false`, those security resources are created as `ClusterRole` and `ClusterRoleBindings`. If the `dedicated` value is set to `true`,  those will be created as `Roles` and `RoleBindings` in the namespace of the operator.
```
Comment (rosemarymarano) [inline docs-source/content/security/rbac.md]: those will be created -> those resources will be created

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md [hunk 23]
```diff
@@ -94,6 +94,23 @@ Example:
 serviceAccount: "weblogic-operator"
 ```
 
+##### `dedicated`
+
+Specifies if this operator will only manage WebLogic Domains in the same namespace in which the operator itself is deployed.  If set to `true`, then the `domainNamespaces` value is ignored.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: WebLogic Domains -> WebLogic domains

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md [hunk 24]
```diff
@@ -94,6 +94,23 @@ Example:
...
+Specifies if this operator will only manage WebLogic Domains in the same namespace in which the operator itself is deployed.  If set to `true`, then the `domainNamespaces` value is ignored. 
+
+Defaults to `false`.
+
+Example:
+```
+dedicated: false
+```
+
+In the `dedicated` mode, the operator does not require permissions to access the cluster-scoped Kubernetes resources such as `CustomResourceDefinitions`, `PersistentVolumes`, and `Namespaces`. In those situations, the operator may skip some of its operations, such as verifying the WebLogic Domain `CustomResoruceDefinition` `domains.weblogic.oracle` (and creating it when it is absent), watching namespace events, and cleaning up `PersistentVolumes` as part of deleting a domain.
```
Comment (rosemarymarano) [inline docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: cluster-scoped Kubernetes resources such as -> cluster-scoped Kubernetes resources, such as
WebLogic Domain -> WebLogic domain

File: operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java (hunk 20 above)
Comment (markxnelson) [reply operator/src/test/java/oracle/kubernetes/operator/StartupControlTest.java]: ok

**Comment by doxiao** [comment]
> some minor edits

Thanks, Rosemary!

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md (hunk 24 above)
Comment (doxiao) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: done

File: docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md (hunk 23 above)
Comment (doxiao) [reply docs-source/content/userguide/managing-operators/using-the-operator/using-helm.md]: done

File: docs-source/content/security/rbac.md (hunk 22 above)
Comment (doxiao) [reply docs-source/content/security/rbac.md]: done

File: docs-source/content/security/openshift.md (hunk 21 above)
Comment (doxiao) [reply docs-source/content/security/openshift.md]: done

**Pull request merged**